Bottom: d4b83b948001ab7b147d92e1a2f7f191cdc30e3c
Top:    bb46541a266307389f09048c4accc6f895ef5351
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-08-25 01:40:33 +0100

HACK: Play with where the GDK version needs to draw crosshair attachments

THESE CHANGES ARE NOT WANTED, JUST A QUICK HACK TO EXPERIMENT WITH


---

diff --git a/src/hid/gtk/gtkhid-gdk.c b/src/hid/gtk/gtkhid-gdk.c
index 5b52b9a..66c5241 100644
--- a/src/hid/gtk/gtkhid-gdk.c
+++ b/src/hid/gtk/gtkhid-gdk.c
@@ -923,6 +923,8 @@ ghid_invalidate_lr (int left, int right, int top, int bottom)
   ghid_invalidate_all ();
 }
 
+void DrawAttached (bool);
+
 void
 ghid_invalidate_all ()
 {
@@ -984,8 +986,10 @@ ghid_invalidate_all ()
 
   hid_expose_callback (&ghid_hid, &region, 0);
   ghid_draw_grid ();
-  if (ghidgui->need_restore_crosshair)
+  if (ghidgui->need_restore_crosshair) {
     RestoreCrosshair (FALSE);
+    DrawAttached (FALSE);
+  }
   ghidgui->need_restore_crosshair = FALSE;
   ghid_screen_update ();
 }
diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index 7cd464a..4d9cb3a 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -228,6 +228,8 @@ ghid_get_coords (const char *msg, int *x, int *y)
     }
 }
 
+void DrawAttached (bool);
+
 gboolean
 ghid_note_event_location (GdkEventButton * ev)
 {
@@ -248,6 +250,7 @@ ghid_note_event_location (GdkEventButton * ev)
   gport->view_x = event_x * gport->zoom + gport->view_x0;
   gport->view_y = event_y * gport->zoom + gport->view_y0;
 
+  DrawAttached (false);
   moved = MoveCrosshairAbsolute (SIDE_X (gport->view_x), 
 				 SIDE_Y (gport->view_y));
   if (moved)
@@ -255,6 +258,7 @@ ghid_note_event_location (GdkEventButton * ev)
       AdjustAttachedObjects ();
       RestoreCrosshair (false);
     }
+  DrawAttached (false);
   ghid_set_cursor_position_labels ();
   return moved;
 }
